src/bytes.rs

```diff
@@ -361,15 +306,26 @@ impl CharIterator {
         // If we still can move forward in the stream, move forwards
         if self.position.offset < self.length {
             let c = self.buffer[self.position.offset];
-            self.seek(SeekMode::SeekCur, 1);
+            if c == Ch('\n') {
```

Comment:
even though we cannot "seek" in a stream, we can unwind a single character. This would be easy, but at a start of the line, we must know the column ending. We store this in line_offsets so we don't need to calculate it.